twine/commands/upload.py

```diff
@@ -58,16 +58,37 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     uploads = [i for i in dists if not i.endswith(".asc")]
     upload_settings.check_repository_url()
     repository_url = cast(str, upload_settings.repository_config["repository"])
+    packages_to_upload = []
+
+    for filename in uploads:
```

Comment:
I really like seeing this behavior extracted from the upload loop, and it gives me more refactoring ideas. My first thought is to pull the whole thing into a separate function, e.g.:

```python
packages_to_upload = _make_packages(uploads, signatures, upload_settings)
```

However, whenever I see this idiom:

```python
new_list = []
for item in other_list:
    new_item = transform(item)
    new_list.append(new_item)
```

I think: can I make this a list comprehension? The answer is usually yes (see: [a good talk](https://www.youtube.com/watch?v=ei71YpmfRX4) from PyCon this year).

In this case, you could use a helper function, e.g.:

```python
packages_to_upload = [
    _make_package(filename, signatures, upload_settings)
    for filename in uploads
]
```

Both of these options approaches would make it easier to unit test.